crates/weaver_resolver/src/registry.rs

```diff
@@ -73,33 +79,44 @@ pub fn resolve_semconv_registry(
 
     all_refs_resolved &= resolve_attribute_references(&mut ureg, attr_catalog);
     all_refs_resolved &= resolve_extends_references(&mut ureg);
+    all_refs_resolved &= resolve_include_constraints(&mut ureg);
 
+    // If the resolution process fails, then we build an error containing all
+    // the unresolved references.
     if !all_refs_resolved {
-        // Process all unresolved references.
-        // An Error::UnresolvedReferences is built and returned.
-        let mut unresolved_refs = vec![];
+        let mut errors = vec![];
         for group in ureg.groups.iter() {
+            // Collect unresolved `extends` references.
             if let Some(extends) = group.group.extends.as_ref() {
-                unresolved_refs.push(UnresolvedReference::ExtendsRef {
+                errors.push(Error::UnresolvedExtendsRef {
                     group_id: group.group.id.clone(),
                     extends_ref: extends.clone(),
                     provenance: group.provenance.clone(),
                 });
             }
+            // Collect unresolved `ref` attributes.
             for attr in group.attributes.iter() {
                 if let AttributeSpec::Ref { r#ref, .. } = &attr.spec {
-                    unresolved_refs.push(UnresolvedReference::AttributeRef {
+                    errors.push(Error::UnresolvedAttributeRef {
                         group_id: group.group.id.clone(),
                         attribute_ref: r#ref.clone(),
                         provenance: group.provenance.clone(),
                     });
                 }
             }
+            // Collect unresolved `include` constraints.
+            for constraint in group.group.constraints.iter() {
+                if let Some(include) = &constraint.include {
+                    errors.push(Error::UnresolvedIncludeRef {
+                        group_id: group.group.id.clone(),
+                        include_ref: include.clone(),
+                        provenance: group.provenance.clone(),
+                    });
+                }
+            }
         }
-        if !unresolved_refs.is_empty() {
-            return Err(Error::UnresolvedReferences {
-                refs: unresolved_refs,
-            });
+        if !errors.is_empty() {
```

Comment:
Nit:  You can probably make a helper function that throws the first item or all errors.  Perhaps even something like "TryFrom".

```rust
if let Ok(error) = errors.try_into() {
  return Err(error);
}
```


And then somewhere:

```rust
impl TryFrom<Vec<Error>> for Error {
  type Error = ErrorConversionFailed;
  fn try_from(value: Vec<Error>) -> Result<Self, Self::Error> {
    if value.is_empty() {
        return Err(ErrorConversionFailed{});
    } else if value.len() == 1 {
       return Err(vec.into_iter().next().unwrap());
    } else {
      return Err(Error::CompoundError(errors));
    }
  }
}
```

Or some such.  COnsider this a non-blocking idea.

Reply:
I did a slightly different version of this suggestion to simplify even further the caller site.

Caller site: 
```rust
handle_errors(errors)?;
```

Function definition.
```rust
/// Handles a list of errors and returns a compound error if the list is not
/// empty or () if the list is empty.
pub fn handle_errors(errors: Vec<Error>) -> Result<(), Error> {
    if errors.is_empty() {
        Ok(())
    } else {
        Err(Error::CompoundError(errors))
    }
}
```